src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -208,6 +205,9 @@ public ChangeLogParser createChangeLogParser() {
     }
 
     public List<BranchSpec> getBranches() {
+        if (gitSCM == null) {
```

Comment:
Is it possible for this case to be prevented entirely? I don't like the idea of the constructor failing because of an invalid form, and would rather the error log be produced when they hit build rather than here.
But if it's unavoidable, disregard this comment.

Reply:
well its a bit difficult. For us, since we want to retain as much as user config, there is no invalid form.